rfcs/0000-rlp-v2.md

```diff
@@ -0,0 +1,1028 @@
+
+  ```yaml
+  gateway_actions:
+  - rules:
+      - paths: ["/toys*"]
+        methods: ["GET"]
+        hosts: ["*.toystore.com"]
+    configurations:
+      - generic_key:
+          descriptor_key: "context.request.http.route_matcher"
+          descriptor_value: "2718701ec9bfd79132e58d92aed722489443094bf9c616e1b74361fe68360f05--toystore/toystore-per-endpoint-per-user/readers"
+      - metadata:
+          descriptor_key: "auth.identity.group"
+          metadata_key:
+            key: "envoy.filters.http.ext_authz"
+            path:
+              - segment:
+                  key: "identity"
+              - segment:
+                  key: "group"
+      - metadata:
+          descriptor_key: "auth.identity.username"
+          metadata_key:
+            key: "envoy.filters.http.ext_authz"
+            path:
+              - segment:
+                  key: "identity"
+              - segment:
+                  key: "username"
+  - rules:
+      - paths: ["/toys*"]
+        methods: ["POST"]
+        hosts: ["*.toystore.com"]
+    configurations:
+      - generic_key:
+          descriptor_key: "context.request.http.route_matcher"
+          descriptor_value: "4f70fc57ad52a2664e3920f373633a9b2b2f4f58f17b39a8d3a3a3485fd91c4d--toystore/toystore-per-endpoint-per-user/writers"
+      - metadata:
+        descriptor_key: "auth.identity.group"
+        metadata_key:
+          key: "envoy.filters.http.ext_authz"
+          path:
+            - segment:
+                key: "identity"
+            - segment:
+                key: "group"
+      - metadata:
+          descriptor_key: "auth.identity.username"
+          metadata_key:
+            key: "envoy.filters.http.ext_authz"
+            path:
+              - segment:
+                  key: "identity"
+              - segment:
+                  key: "username"
+  ```
+
+  ```yaml
+  limits:
+  - conditions:
+      - context.request.http.route_matcher == "2718701ec9bfd79132e58d92aed722489443094bf9c616e1b74361fe68360f05--toystore/toystore-per-endpoint-per-user/readers"
+      - auth.identity.group != "admin"
+    variables:
+      - auth.identity.username
+    max_value: 50
+    seconds: 60
+    namespace: "*.toystore.com"
+  - conditions:
+      - context.request.http.route_matcher == "4f70fc57ad52a2664e3920f373633a9b2b2f4f58f17b39a8d3a3a3485fd91c4d--toystore/toystore-per-endpoint-per-user/writers"
+      - auth.identity.group != "admin"
+    variables:
+      - auth.identity.username
+    max_value: 5
+    seconds: 60
+    namespace: "*.toystore.com"
+  - conditions:
+      - context.request.http.route_matcher == "4f70fc57ad52a2664e3920f373633a9b2b2f4f58f17b39a8d3a3a3485fd91c4d--toystore/toystore-per-endpoint-per-user/writers"
+      - auth.identity.group != "admin"
+    variables:
+      - auth.identity.username
+    max_value: 100
+    seconds: 43200 # 12 hours
+    namespace: "*.toystore.com"
+  ```
+</details>
+
+#### Example 3. Route filtering using `when` conditions
```

Comment:
I'd be tempted to _not_ support this for now and force users to fallback to `Example 4` instead… this can always be introduced later _when_ the need arises.

Reply:
I don't think that's possible. That would assume the HTTPRoute can always be changed (by imposition of waiting to write a policy that targets a specific match). There's also the case of targeting Gateways and yet writing conditions based on attributed of the HTTP request.